lib/editor/components/timetable/TimetableEditor.js

```diff
@@ -220,11 +220,22 @@ export default class TimetableEditor extends Component<Props, State> {
           cumulativeTravelTime += +halt.flexDefaultTravelTime
 
           // $FlowFixMe Flow doesn't recognize our check for stopId property?
-          objectPath.set(newRow, `stopTimes.${i}.stopId`, patternHaltIsLocation(halt) ? halt.locationId : halt.locationGroupId)
+          const haltId = patternHaltIsLocation(halt) ? halt.locationId : halt.locationGroupId
+          const nextHalt = i + 1 !== patternHalts.length ? patternHalts[i + 1] : null
+          let nextHaltId
```

Comment:
Instead of storing the id separately, can't we just create/use a method that returns the id of a halt no matter the halt type?

Reply:
I think I agree, https://github.com/ibi-group/datatools-ui/pull/930/commits/e75153f1e1e68b66f7467976b34869a461ec3462 looks a lot cleaner